google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/SpannerDialect.java

```diff
@@ -369,6 +369,19 @@ public String getAddPrimaryKeyConstraintString(String constraintName) {
     throw new UnsupportedOperationException("Cannot add primary key constraint in Cloud Spanner.");
   }
 
+  /* Query Hints */
+
+  @Override
+  public String getQueryHintString(String query, List<String> hintsList) {
+    String formattedHints = QueryHintProcessor.formatQueryHints(hintsList);
+    return getQueryHintString(query, formattedHints);
+  }
+
+  @Override
+  public String getQueryHintString(String query, String hints) {
+    return hints + " " + query;
```

Comment:
Do we need to check if the `hints` string is null or empty?

Reply:
Nice catch; empty yes. I'm pretty sure we don't need to check null but maybe it can't hurt to add it in...